src/Tokens.ts

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable no-use-before-define */
+export type Token = (Tokens.Space
+    | Tokens.Code
+    | Tokens.Heading
+    | Tokens.Table
+    | Tokens.Hr
+    | Tokens.Blockquote
+    | Tokens.List
+    | Tokens.ListItem
+    | Tokens.Paragraph
+    | Tokens.HTML
+    | Tokens.Text
+    | Tokens.Def
+    | Tokens.Escape
+    | Tokens.Tag
+    | Tokens.Image
+    | Tokens.Link
+    | Tokens.Strong
+    | Tokens.Em
+    | Tokens.Codespan
+    | Tokens.Br
+    | Tokens.Del) & { loose?: boolean, tokens?: Token[] };
+
+export namespace Tokens {
+    export interface Space {
+        type: 'space';
+        raw: string;
+    }
+
+    export interface Code {
+        type: 'code';
+        raw: string;
+        codeBlockStyle?: 'indented' | undefined;
+        lang?: string | undefined;
```

Comment:
My understanding is the question mark already identifies this property as optional.

```suggestion
        codeBlockStyle?: 'indented';
        lang?: string;
```

Is this equivalent? If so, I would prefer this simplified notation rather than mixing both the `?` and ` | undefined`

Reply:
The mark markers the property as options, so you can omit it when declaring an object literal, which results in undefined if you try to access it, while the explicit undefined type allows to explicitly assign the value undefined to the property which happens for example in the defaults.